### PR TITLE
Simplify CodeQL setup now that the last two releases have semver bundles

### PR DESCRIPTION
**Pull request by henrymercer**
Now that the last two CLI versions have semantically versioned bundles, we no longer need to handle the case where the default version feature flags specify a version that doesn't have a semantically versioned bundle.  As a result, we can reduce the complexity of the setup code!

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: src/setup-codeql.ts [hunk 1]
```diff
@@ -535,16 +480,8 @@ export async function getCodeQLSource(
...
-      tagName = await findCodeQLBundleTagDotcomOnly(cliVersion, logger);
-    } else if (!tagName) {
-      throw new Error(
-        `Could not obtain the requested version (${humanReadableVersion}) of the CodeQL tools ` +
-          "since we could not compute the tag name."
-      );
-    }
     url = await getCodeQLBundleDownloadURL(
-      tagName,
+      tagName!,
```
Comment (angelapwen) [inline src/setup-codeql.ts:477]: TIL about this operator 🤯

**Review by angelapwen — APPROVED** [review]
🥳

File: src/setup-codeql.ts (hunk 1 above)
Comment (henrymercer) [reply src/setup-codeql.ts:477]: It's not the best sign when you have to use it, but here it didn't seem worth refactoring the types :)

File: src/setup-codeql.ts [hunk 2]
```diff
@@ -475,12 +421,11 @@
   }
 
   // Fall back to matching `0.0.0-<bundleVersion>`.
-  if (!codeqlFolder && (cliVersion || tagName)) {
-    if (cliVersion || tagName) {
+  if (!codeqlFolder && tagName) {
+    if (tagName) {
```
[the remark this answers is not in the document]
Comment (Spinike) [reply src/setup-codeql.ts]: - [ ] - - [x] @b4mboo]()

[the remark this answers is not in the document]
Comment (Spinike) [reply src/setup-codeql.ts]: [image]()

**Pull request merged**